consul/base.py

```diff
@@ -138,21 +138,33 @@ def callback(
         is_json=False,
         index=False,
         one=False,
-        allow_404=True):
+        allow_404=True,
+        decode=None,
+        is_id=False):
```

Comment:
Hi there, thanks for this patch!

I've been a bit cautious about beefing up and switching to this callback convenience, as it obscures what processing will take. The trade of course is it cuts down on verbosity, but more importantly, ensures consistent handling of responses.

If we can get the api right for it, it could help reduce the downside of obscuring things.

Is it possible to add a docstring that describes what effect each of the options will have on the response?

Something I also think will help is that for options that can only take effect, when another options is set, is to move them to sub-options. e.g. `decode` and `is_id` can only be used with `is_json`. perhaps `is_json` could take a dict, which has the option `decode` and `is_id` options?

What do you think?